src/lambda_handler.py

```diff
@@ -1,89 +1,207 @@
+            if self.cvss_score == 10:
+                self.severity = 4
+            elif self.cvss_score >= 7.0:
+                self.severity = 3
+            elif self.cvss_score >= 4.0:
+                self.severity = 2
+            else:
+                self.severity = 1
+        elif self.cvss_version in ["3.0", "3.1"]:
+            if self.cvss_score >= 9.0:
+                self.severity = 4
+            elif self.cvss_score >= 7.0:
+                self.severity = 3
+            elif self.cvss_score >= 4.0:
+                self.severity = 2
+            else:
+                self.severity = 1
+        await super().save(*args, **kwargs)
 
-    task = event.get("task", None)
-    logging.error(error_msg, task)
-    failed_task(result, error_msg % task)
 
-    return result
+async def process_nvd(cve) -> str:
+    """Add the provided CVE to the database and return its id."""
+    # Fill fields for the CVE Documents from the given JSON files
+    cve_id = cve["cve"]["CVE_data_meta"]["ID"]
+    # Reject or remove CVEs that don't have baseMetricV2 or baseMetricV3 CVSS data
+    if not any(k in cve["impact"] for k in ["baseMetricV2", "baseMetricV3"]):
+        """If the CVE is in the database, it needs to be deleted from the database"""
 
+        if await CVEDoc.find_one(CVEDoc.id == cve_id):
+            print("x", end="")
+            remove_cve_doc = CVEDoc(id=cve_id)
+            await remove_cve_doc.delete()
 
-def task_cowsay(event) -> dict[str, Union[Optional[str], bool]]:
-    """Generate an output message using the provided information."""
-    result: dict[str, Union[Optional[str], bool]] = {"message": None, "success": True}
+            return cve_id
+
+        else:
+            print("x", end="")
+            return cve_id
 
-    character: str = event.get("character", "tux")
-    if character not in cowsay.characters.CHARS.keys():
-        error_msg = 'Character "%s" is not valid.'
-        logging.error(error_msg, character)
-        failed_task(result, error_msg % character)
     else:
-        contents: str = event.get("contents", "Hello from AWS Lambda!")
-        logger.info(
-            'Creating output using "%s" with contents "%s"', character, contents
+        version = "V3" if "baseMetricV3" in cve["impact"] else "V2"
+        cvss_base_score = cve["impact"]["baseMetric" + version]["cvss" + version][
+            "baseScore"
+        ]
+        cvss_version_temp = cve["impact"]["baseMetric" + version]["cvss" + version][
+            "version"
+        ]
+
+        # Fill document fields with CVE data
+        print(".", end="")
+
+        print("Adding CVE: ", cve_id, " to database")
+
+        cve_doc = CVEDoc(
+            id=cve_id, cvss_score=float(cvss_base_score), cvss_version=cvss_version_temp
         )
-        result["message"] = cowsay.get_output_string(character, contents)
 
-    return result
+        if not cve_id:
+            raise ValueError("JSON does not look like valid CISA CVE data.")
```

Comment:
Should this be earlier in the function, say just after you set `cve_id` in line 75?

Reply:
This has been addressed in the latest push. https://github.com/cisagov/cyhy-nvdsync-lambda/pull/2/commits/1220b72e11ee456746d6f29feab755e09d089c8d